releasenotes.rb

```diff
@@ -3,9 +3,35 @@
 
 require "thor"
 require "git"
+require "octokit"
+require "logger"
 require "erb"
 require "ostruct"
 
+class GitHubCommitTags
+  attr_reader :client, :repository, :logger
+
+  def initialize(access_token:, repository:, logger: Logger.new(STDERR))
+    @client = Octokit::Client.new(access_token: access_token)
+    @repository = repository
+    @logger = logger
+  end
+
+  def special_deploy_requirements?(sha)
+    search_results = client.search_issues("#{sha} repo:#{repository} is:merged")
+
+    if search_results.total_count != 1
+      logger.warn("SHA #{sha} had #{search_results.total_count} issues, not just one; ignoring.")
```

Comment:
Inside of the git repository, there's no direct connection between a commit and the PR it came from - PRs are an artifact of GitHub. The only way to find what PR a commit came from is through the GH search.

If we do a search for a commit and find no corresponding pull requests, then that means the commit was pushed to the branch without a PR. If we find *more* than two PRs for the same commit, then **I have no idea what has happened**, so I wanted to warn about it in a big way.

@markzalar's case is more annoying. In general, I think we'd want to know when a commit gets pushed to `deploy` without a PR, but there will always be at least one: the commit that merges `master` into `deploy`. We could decide that we aren't worried about that, in which case we'd change this to just `> 1`.